File: mo-common/mo-common-oss/src/main/java/ink/ckx/mo/common/oss/config/AliyunOssConfig.java
package ink.ckx.mo.common.oss.config;

import com.aliyun.oss.OSS;
import com.aliyun.oss.OSSClientBuilder;
import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.stereotype.Component;

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/14
 */
@ConfigurationProperties(prefix = "oss.aliyun")
@Component
@Data
public class AliyunOssConfig {

    /**
     * 访问凭据
     */
    private String accessKeyId;

    /**
     * 凭据密钥
     */
    private String accessKeySecret;

    /**
     * 服务 Endpoint
     */
    private String endpoint;

    /**
     * 存储桶名称
     */
    private String bucketName;

    @Bean
    public OSS oSS() {
        return new OSSClientBuilder().build(endpoint, accessKeyId, accessKeySecret);
    }
}

File: mo-common/mo-common-oss/src/main/kotlin/ink/ckx/mo/common/oss/config/AliyunOssConfig.kt
package ink.ckx.mo.common.oss.config

import com.aliyun.oss.OSS
import com.aliyun.oss.OSSClientBuilder
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/14
 */
@Configuration
class AliyunOssConfig {

    @Bean
    @ConfigurationProperties(prefix = "oss.aliyun")
    fun aliyunOssConfigProperties(): AliyunOssConfigProperties {
        return AliyunOssConfigProperties()
    }

    @Bean
    fun oSS(properties: AliyunOssConfigProperties): OSS {
        return OSSClientBuilder().build(properties.endpoint, properties.accessKeyId, properties.accessKeySecret)
    }
}

class AliyunOssConfigProperties {

    /**
     * 访问凭据
     */
    var accessKeyId: String? = null

    /**
     * 凭据密钥
     */
    var accessKeySecret: String? = null

    /**
     * 服务 Endpoint
     */
    var endpoint: String? = null

    /**
     * 存储桶名称
     */
    var bucketName: String? = null
}